### ISSUE TITLE
External HTTPS requests with Python urllib2 not working

### ISSUE THREAD

**Issue by mihai-tiriplica-hs** [istio/istio#14393]
(This is used to report product bugs, please visit <https://discuss.istio.io> for questions on using Istio)

**Bug description**

I'm trying to make HTTPS calls to an external resource using urllib2 in Python 2.7.6 and they fail with "connection reset by peer" (error code 104). When I try the same request with Python requests, urllib3 or with curl the request works as expected.

**Expected behavior**

Calls with urllib2 work as expected. It would be nice to at least find the reason for this.

**Version (include the output of `istioctl version --remote` and `kubectl version`)**

Istio 1.0.7
Kubernetes 1.11.6

**How was Istio installed?**

Helm templates and kubectl apply

**Environment where bug was observed (cloud vendor, OS, etc)**

K8s on AWS, python 2.7.6

**Affected product area (please put an X in all that apply)**

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[x] Security
[ ] Test and Release
[ ] User Experience

Additionally, please consider attaching a [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) by attaching
the dump file to this issue.

**Comment by mihai-tiriplica-hs** [comment]
I tried Python 2.7.16 and everything works but I just can't pin point the problem. I'm pretty sure it's something related to how SSL connections are handled but I can't put the pieces together.

**Comment by howardjohn** [comment]
Istio 1.0.7 by default will block external calls, maybe that is the issue?

On Istio 1.2, I ran

```
import urllib2
response = urllib2.urlopen('http://python.org/')
html = response.read()
```

and the call succeeded. Is this still an issue?

**Comment by mihai-tiriplica-hs** [comment]
The issue was specifically with the Python version. I tried upgrading python and it worked. For me it's not an issue right now since I upgraded Python but maybe others get in the same situation.

**Comment by howardjohn** [comment]
I ran on python 2.7.6,the original version in the issue and could not reproduce. Unless someone can reproduce this I will assume it's fixed since this was reported for Istio 1.0. thanks!!
